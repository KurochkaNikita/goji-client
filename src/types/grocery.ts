export type TGroceryListItem = {
  id: string;
  completed: boolean;
  title: string;
  amount: number;
};

export type TGroceryList = {
  id: string;
  name: string;
  list: TGroceryListItem[];
};

export type ListItemForm = {
  title: string;
  amount?: number;
};
