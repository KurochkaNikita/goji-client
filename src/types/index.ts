export type TNextPage<P = never> = {
  params: Record<string, string> & P;
  searchParams: Record<string, string>;
};

export type TBreadcrumbs = {
  link: string;
  label: string;
};
