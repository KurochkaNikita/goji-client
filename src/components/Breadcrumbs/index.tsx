import React from "react";
import Link from "next/link";
import MUIBreadcrumbs from "@mui/material/Breadcrumbs";
import Grid from "@mui/material/Grid2";
import Typography from "@mui/material/Typography";
import { TBreadcrumbs } from "types";

interface IBreadcrumbs {
  currentPage: string;
  items?: TBreadcrumbs[];
}

function Breadcrumbs(props: IBreadcrumbs) {
  const { items = [], currentPage } = props;
  return (
    <Grid size={12}>
      <MUIBreadcrumbs aria-label="breadcrumb">
        {items.map((item) => (
          <Link color="inherit" href={item.link} key={item.link}>
            {item.label}
          </Link>
        ))}
        <Typography sx={{ color: "text.primary" }}>{currentPage}</Typography>
      </MUIBreadcrumbs>
    </Grid>
  );
}

export default Breadcrumbs;
