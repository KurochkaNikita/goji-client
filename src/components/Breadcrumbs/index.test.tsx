import { render } from "@testing-library/react";
import { TBreadcrumbs } from "types";

import Breadcrumbs from "./index";

describe("Breadcrumbs", () => {
  it("Snapshot with empty items", () => {
    const items = [] as TBreadcrumbs[];
    const { container } = render(
      <Breadcrumbs items={items} currentPage="here" />
    );
    expect(container).toMatchSnapshot();
  });

  it("Snapshot with 2 items", () => {
    const items: TBreadcrumbs[] = [
      { link: "/1", label: "1" },
      { link: "/2", label: "2" },
    ];
    const { container } = render(
      <Breadcrumbs items={items} currentPage="here" />
    );
    expect(container).toMatchSnapshot();
  });

  it("Snapshot without items", () => {
    const { container } = render(<Breadcrumbs currentPage="here" />);
    expect(container).toMatchSnapshot();
  });
});
