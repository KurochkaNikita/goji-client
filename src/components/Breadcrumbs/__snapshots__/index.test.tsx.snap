// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Snapshot with 2 items 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            color="inherit"
            href="/1"
          >
            1
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            color="inherit"
            href="/2"
          >
            2
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
          >
            here
          </p>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;

exports[`Breadcrumbs Snapshot with empty items 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
          >
            here
          </p>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;

exports[`Breadcrumbs Snapshot without items 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
          >
            here
          </p>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;
